content/docs/guides/continuous-delivery/pulumi-kubernetes-operator.md

```diff
@@ -0,0 +1,318 @@
+out CI/CD and automation systems into your clusters, creating native support to manage your infrastructure alonside your Kubernetes workloads.
+
+To work with the operator, we'll need to follow these steps.
+
+- [Deploy the Pulumi Kubernetes Operator](#deploy-the-pulumi-kubernetes-operator)
+- [Create a Stack CustomResource](#create-a-stack-customresource)
+
+[k8s-ext-pattern]: https://kubernetes.io/docs/concepts/extend-kubernetes/operator/
+[stack]: {{< relref "/docs/intro/concepts/stack" >}}
+
+## Deploy the Pulumi Kubernetes Operator
+
+The operator is composed of:
+
+- A ServiceAccount for identity,
+- A Role and RoleBinding to the ServiceAccount for RBAC, and
+- A Deployment to manage the controller.
+
+To create the operator, choose an installation preference using Pulumi
+with a supported programming language or `kubectl` with YAML.
+
+- [Installing the Operator with Pulumi in Typescript, Python, C#, and Go][use-pulumi]
+- [Installing the Operator with kubectl][use-kubectl]
+
+[use-pulumi]: https://github.com/pulumi/pulumi-kubernetes-operator#using-pulumi
+[use-kubectl]: https://github.com/pulumi/pulumi-kubernetes-operator#using-kubectl
+
+When launched, the operator invokes the `StackController` to manages updates to
+`Stack` CustomResources created, updated, or deleted in Kubernetes.
+
+These updates are run in the form of reconcilation loops that attempt to update a Stack until success
+is reached for the Git commit SHA provided, also known as the `desired state`.
+
+## Create a Stack CustomResource
+
+The `Stack` CustomResource (CR) encapsulates a Pulumi project that creates any set of
+infrastructure resources such as cloud VMs, object storage, Kubernetes
+clusters, or Kubernetes workloads through API resources.
+
+The Stack uses an existing Git repo, and checks out a specific Git commit SHA of the repo to deploy a `pulumi up`.
+
+In the example below, we're creating a Stack for an existing Pulumi project that provisions
+a simple [NGINX][nginx-stack] Deployment in Kubernetes.
+
+When the Stack is processed and deployed by the operator, NGINX will be created
+in the same cluster as the operator. This is because the NGINX Pulumi program does not
+explicitly use a [Kubernetes Provider resource][k8s-provider], and the Operator
+makes its ServiceAccount credentials available to Stacks that rely on
+the [default, ambient kubeconfig credentials][default-kubeconfig].
+
+The role permissions for the operator can be adjusted to control what in-cluster API resources are allowed.
+
+[nginx-stack]: https://github.com/metral/pulumi-nginx/blob/master/index.ts
+[k8s-provider]: {{< relref "/docs/reference/pkg/kubernetes/provider" >}}
+[default-kubeconfig]: {{< relref "/docs/intro/cloud-providers/kubernetes/setup#steps" >}}
+
+Choose your preferred language below, or check out [Create Pulumi Stacks using kubectl][stacks-use-kubectl].
+
+{{< chooser language "typescript,python,go,csharp" >}}
+{{% choosable language typescript %}}
+
+```typescript
```

Comment:
Per my usual pet peeve - would love to highlight that this can be done with YAML as well!

Reply:
(though this can be a follow-up thing - definitely not blocking getting this content out)